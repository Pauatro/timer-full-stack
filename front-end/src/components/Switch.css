.toggle-switch-container{
    height: 100%;
    flex-grow: 1;
    margin-right: 2.5%;
    margin-left: 2.5%;
}

.toggle-switch {
    position: relative;
    height: 100%;
    width: 100%;
}

.toggle-switch-checkbox {
    display: none;
}

.toggle-switch-label {
    display: block;
    overflow: hidden;
    cursor: pointer;
    border-radius: 2vh;
    height: 100%;
    width: 100%; 
}

.toggle-switch-inner {
    display: block;
    width: 100%;
    height: 100%;
    background-color: var(--switch-primary-color);
    overflow: hidden;
}

.toggle-switch-inner:before,
.toggle-switch-inner:after {
  display: block;
  float: left;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  background-color: var(--switch-secondary-color);
  color: var(--switch-secondary-color);
  content: "N";
}

.toggle-switch-switch {
    display: block;
    position: absolute;
    width: 42.5%;
    height: 80%;
    margin: 5%;
    background: var(--switch-primary-color);
    top: 0;
    bottom: 0;
    right: 47.5%;
    border-radius: 2vh;
    transition: all 0.3s ease-in 0s;
}

.toggle-switch-checkbox:checked  + .toggle-switch-label > .toggle-switch-switch{
    right: 0px;
}