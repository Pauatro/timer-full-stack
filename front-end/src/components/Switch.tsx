import React, { useContext, useState } from 'react';
import { ThemeContext } from '../modules/theme/theme.context';
import './Switch.css'

interface ISwitchProps {
    name: string;
}

const Switch = (props: ISwitchProps) => {

    const { switchTheme } = useContext(ThemeContext)
    const [ checked, setChecked ] = useState(true)

    const switchChecked = ()=>{
        setChecked(!checked)
        switchTheme()
    }

    return (
        <div className="toggle-switch-container">
            <div className="toggle-switch">
                <input
                    onChange = {switchChecked}
                    type="checkbox"
                    className="toggle-switch-checkbox"
                    name={props.name}
                    id={props.name}
                    checked = {checked}
                />
                <label className="toggle-switch-label" htmlFor={props.name}>
                    <span className="toggle-switch-inner" />
                    <span className="toggle-switch-switch" />
                </label>
            </div>
        </div>
    )
}

export default Switch