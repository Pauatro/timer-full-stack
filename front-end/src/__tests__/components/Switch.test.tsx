import React from 'react';
import renderer from 'react-test-renderer';
import Switch from '../../components/Switch';

it('renders correctly', () => {
  const tree = renderer
    .create(<Switch />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});