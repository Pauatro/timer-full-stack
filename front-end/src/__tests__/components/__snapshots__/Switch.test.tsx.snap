// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="toggle-switch-container"
>
  <div
    className="toggle-switch"
  >
    <input
      checked={true}
      className="toggle-switch-checkbox"
      id="theme-switch"
      name="theme-switch"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="toggle-switch-label"
      htmlFor="theme-switch"
    >
      <span
        className="toggle-switch-inner"
      />
      <span
        className="toggle-switch-switch"
      />
    </label>
  </div>
</div>
`;
